### PR TITLE
updated Makefile

### PR DESCRIPTION
**Pull request by YaswanthDasamandam**
Automating the makefile per #13

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,24 +1,62 @@
...
+    if match:
+        target, help = match.groups()
+        output.append("%-10s %s" % (target, help))
+# Sort the output in alphanumeric order
+output.sort()
+# Print the help result
+print('\n'.join(output))
+endef
+export PRINT_HELP_PYSCRIPT # End of python section
+# Define target for printing help message
```
Comment (pydanny) [inline Makefile:23]: For the sake of clarity, add line breaks between sections.

```suggestion

# Define target for printing help message
```

File: Makefile [hunk 2]
```diff
@@ -1,24 +1,62 @@
...
+help:
+	@python -c "$$PRINT_HELP_PYSCRIPT" < $(MAKEFILE_LIST)
+# Define variable for GitHub API URL
+GH_API = https://api.github.com/repos/pydanny/listo/releases/latest
+# Define variable for jq command
+JQ = jq -r
+# Define target for generating changelog
+changelog:  ## Install gh cli and jq first
```
Comment (pydanny) [inline Makefile:32]: For the sake of clarity, add line breaks between sections.

```suggestion

changelog:  ## Install gh cli and jq first
```

File: Makefile [hunk 3]
```diff
@@ -1,24 +1,62 @@
...
   -H "X-GitHub-Api-Version: 2022-11-28" \
-  /repos/pydanny/listo/releases/latest > changelog.json
-	jq  -r '.tag_name' changelog.json
-	jq  -r '.body' changelog.json
-
-lint:
+  $(GH_API) > changelog.json
+	$(JQ) '.tag_name' changelog.json
+	$(JQ) '.body' changelog.json
+# Define target for linting code
```
Comment (pydanny) [inline Makefile:40]: For the sake of clarity, add line breaks between sections.

```suggestion

# Define target for linting code
```

File: Makefile [hunk 4]
```diff
@@ -1,24 +1,62 @@
...
+lint: ## Lint code with black and ruff
 	black .
 	ruff check . --fix
-
+# Define target for testing code
```
Comment (pydanny) [inline Makefile:45]: For the sake of clarity, add line breaks between sections.

```suggestion

# Define target for testing code
```

File: Makefile [hunk 5]
```diff
@@ -1,24 +1,62 @@
...
+test: ## Test code with coverage and pytest
+	coverage run -m pytest .
+	coverage report -m
+	coverage html
+# Define target for tagging version
```
Comment (pydanny) [inline Makefile:51]: For the sake of clarity, add line breaks between sections.

```suggestion

# Define target for tagging version
```

File: Makefile [hunk 6]
```diff
@@ -1,24 +1,62 @@
...
 VERSION=$(shell grep -m 1 version pyproject.toml | tr -s ' ' | tr -d '"' | tr -d "'" | cut -d' ' -f3)
-
-tag:
+tag: ## Tag version with git
 	echo "Tagging version $(VERSION)"
 	git tag -a $(VERSION) -m "Creating version $(VERSION)"
 	git push origin $(VERSION)
+# Define phony target for executing all targets
```
Comment (pydanny) [inline Makefile:58]: For the sake of clarity, add line breaks between sections.

```suggestion

# Define phony target for executing all targets
```

File: Makefile [hunk 7]
```diff
@@ -1,24 +1,62 @@
...
+all: changelog lint test tag ## Execute all targets
+# Define phony target for cleaning up files
```
Comment (pydanny) [inline Makefile:61]: For the sake of clarity, add line breaks between sections.

```suggestion

# Define phony target for cleaning up files
```

File: Makefile [hunk 8]
```diff
@@ -1,24 +1,62 @@
...
+.PHONY: clean
+clean: ## Clean up files generated by makefile
+	rm -f changelog.json
+	rm -rf htmlcov
+	rm -f .gitignore
```
Comment (pydanny) [inline Makefile]: The .gitignore is not something we are modifying with the makefile

```suggestion
```

File: Makefile [hunk 9]
```diff
@@ -1,24 +1,62 @@
...
+# Declare phony targets
```
Comment (pydanny) [inline Makefile:67]: For the sake of clarity, add line breaks between sections.

```suggestion

# Declare phony targets
```

**Review by pydanny — CHANGES_REQUESTED** [review]
Wow, thanks for submitting the first contributor PR! Good work on providing something function and useful. 😄 

I left some comments and suggestions.

**Review by pydanny — APPROVED** [review]
Thanks for updating everything! Let's merge it in!

**Comment by YaswanthDasamandam** [comment]
Glad to be of help

**Pull request merged**